R-package/R/lgb.Booster.R

```diff
@@ -770,15 +784,13 @@ Booster <- R6::R6Class(
 #'    )
 #' )
 #' }
-#' @importFrom utils modifyList
+#' @importFrom utils modifyList head
 #' @export
 predict.lgb.Booster <- function(object,
                                 newdata,
+                                type = c("link", "response", "raw", "leaf", "contrib"),
                                 start_iteration = NULL,
                                 num_iteration = NULL,
-                                rawscore = FALSE,
-                                predleaf = FALSE,
```

Comment:
Down in the section where this function catches arguments that fall into `...`

https://github.com/microsoft/LightGBM/blob/f715645725bcfd7750b9f0a535ef9fd67454eb16/R-package/R/lgb.Booster.R#L822-L824

Please add the following:

```r
if (isTRUE(additional_params[["rawscore"]])) {
    stop("Argument 'rawscore' is no longer supported. Use type = 'raw' instead.")
}
if (isTRUE(additional_params[["predleaf"]])) {
    stop("Argument 'predleaf' is no longer supported. Use type = 'leaf' instead.")
}
if (isTRUE(additional_params[["predcontrib"]])) {
    stop("Argument 'predcontrib' is no longer supported. Use type = 'contrib' instead.")
}
```

I'm ok with breaking users' code in the next release in exchange for making the package's interface more compatible with other packages for modeling in R, but I think we should provide specific, actionable error messages when possible to reduce the effort required for affected users to alter their code.